api/infrastructure/parallelcluster-api.yaml

```diff
@@ -0,0 +1,246 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: 'Template for the ParallelCluster API'
+
+Parameters:
+  EnableMultiRegionSupport:
+    Description: When set to true it enables the API to control clusters in all regions
+    Type: String
+    Default: false
+    AllowedValues:
+      - true
+      - false
+
+  EcrImageUri:
+    Description: When specified use this image for the Lambda function and skip the import phase
+    Type: String
+    Default: ''
+
+  ApiDefinitionS3Uri:
+    Description: S3 URI of the ParallelCluster API spec
+    Type: String
+    Default: uri_of_pc_api # TODO change with actual URI
+
+  CustomDomainName:
+    Description: When specified, the custom domain name of the ParallelCluster API
+    Type: String
+    Default: ''
+
+  CustomDomainCertificate:
+    Description: When specified, the ARN of the certificate for custom domain name of the ParallelCluster API
+    Type: String
+    Default: ''
+
+  CustomDomainHostedZoneId:
+    Description: When specified, the id of the Hosted Zone of the custom domain name of the ParallelCluster API
+    Type: String
+    Default: ''
+
+  PublicEcrImageUri:
+    Description: When specified, the URI of the Docker image for the Lambda of the ParallelCluster API
+    Type: String
+    Default: uri_of_public_ecr_image # TODO change with actual URI
+
+  ParallelClusterVersion:
```

Comment:
I'm wondering if we should move this to a Mapping rather than a Parameter so that the user cannot change this

Reply:
Agreed, the `ParallelClusterVersion` should be a "constant", not a parameter that the customer should specify. Question, should we also make the following a Mapping?
* `ApiDefinitionS3Uri`
* `PublicEcrImageUri`

Reply:
I believe that those will be needed as parameter to test the develop version